kclvm/parser/src/parser/ty.rs

```diff
@@ -219,7 +219,44 @@ impl<'a> Parser<'a> {
                 self.sess.struct_token_loc(token, self.prev_token),
             ));
         }
+        // (type) -> type
+        else if let TokenKind::OpenDelim(DelimToken::Paren) = self.token.kind {
+            // Parse function type start with '('
+            self.bump_token(TokenKind::OpenDelim(DelimToken::Paren));
 
+            let mut params_type = vec![];
+            // Parse all the params type until the params list end ')'
+            while self.token.kind != TokenKind::CloseDelim(DelimToken::Paren) {
```

Comment:
Attention should be paid to the error recovery of parse in this area (Avoid dead loop.). Ref: https://github.com/kcl-lang/kcl/issues/420